### PR TITLE
Pass $form_id to give_currency_filter.

### LINKED ISSUE #2311: Add $form_id support to give_currency_filter
## Issue Overview

For ref: https://github.com/WordImpress/Give/pull/2290

## Todos
- [ ] Tests
- [ ] Documentation

### PR DESCRIPTION
**Pull request by emgk**
## Description
Allowing form_id to `give_currency_filter` function will filter the currency by form based.
this PR will fix #2311.

## How Has This Been Tested?
- Manually in the add-on.

## Types of changes
New feature (non-breaking change which adds functionality) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/formatting.php [hunk 1]
```diff
@@ -544,16 +544,17 @@ function give_format_decimal( $amount, $dp = false, $sanitize = true ) {
...
- * @param bool   $decode_currency Whether to decode the currency HTML format or not.
+ * @param string         $price           The donation amount.
+ * @param string         $currency        The currency code.
+ * @param bool           $decode_currency Whether to decode the currency HTML format or not.
+ * @param string|integer $form_id         Donation Form ID.
  *
  * @return mixed|string
  */
-function give_currency_filter( $price = '', $currency = '', $decode_currency = false ) {
+function give_currency_filter( $price = '', $currency = '', $decode_currency = false, $form_id = '' ) {
```
Comment (ravinderk) [inline includes/formatting.php]: @emgk Can we convert param into single array with backward compatibility

Comment (emgk) [reply includes/formatting.php]: Sure, I will do it on Monday.  @ravinderk

**Comment by DevinWalker** [comment]
Can we resolve the conflict here?

**Comment by emgk** [comment]
@DevinWalker @ravinderk I have resolved conflict and converted parameters into the array with backward compatibility.

File: includes/currency-functions.php [hunk 2]
```diff
@@ -513,16 +513,41 @@ function give_get_currency_name( $currency_code ) {
...
-		$currency_code = give_get_currency();
+	// Bailout.
+	if ( empty( $price ) || ( ! is_numeric( $price ) && ! is_string( $price ) ) ) {
+		return $price;
+	}
+
+	// Get functions arguments.
+	$func_args = func_get_args();
+
+	if ( isset( $func_args[1] ) && ( is_string( $func_args[1] ) || is_numeric( $func_args[1] ) ) ) {
```
Comment (ravinderk) [inline includes/currency-functions.php]: @emgk `$func_args[1]` is currency_code why are you check this as `is_numeric`

Comment (emgk) [reply includes/currency-functions.php]: @ravinderk thanks, I have removed `is_numeric` checking on this.

File: includes/currency-functions.php [hunk 3]
```diff
@@ -588,11 +609,11 @@ function give_currency_filter( $price = '', $currency_code = '', $decode_currenc
...
 	 */
 	$formatted = apply_filters(
-		'give_' . strtolower( $currency_code ) . "_currency_filter_{$position}",
+		'give_' . strtolower( $args['currency_code'] ) . "_currency_filter_{$position}",
 		$formatted,
-		$currency_code,
+		$args['currency_code'],
 		$price,
-		$decode_currency
+		$args
```
Comment (ravinderk) [inline includes/currency-functions.php]: @emgk I pass all argument to this filter, so it will be useful for you.

Comment (emgk) [reply includes/currency-functions.php]: Yes, that will be useful, Thanks @ravinderk

**Pull request merged**